### DISCUSSION TITLE
Can't get firmware for Robin clone V2.1

### DISCUSSION THREAD

**Discussion by cyb1337** [Guilouz/Klipper-Flsun-Speeder-Pad#50] [Q&A] [answered]
pi@speeder-pad:~/klipper$ make clean
pi@speeder-pad:~/klipper$ make
  Creating symbolic link out/board
  Building out/autoconf.h
  Compiling out/src/sched.o
  Compiling out/src/command.o
  Compiling out/src/basecmd.o
  Compiling out/src/debugcmds.o
  Compiling out/src/initial_pins.o
  Compiling out/src/gpiocmds.o
  Compiling out/src/stepper.o
  Compiling out/src/endstop.o
  Compiling out/src/trsync.o
  Compiling out/src/adccmds.o
  Compiling out/src/spicmds.o
  Compiling out/src/thermocouple.o
  Compiling out/src/i2ccmds.o
  Compiling out/src/pwmcmds.o
  Compiling out/src/spi_software.o
  Compiling out/src/sensor_adxl345.o
  Compiling out/src/sensor_angle.o
  Compiling out/src/sensor_mpu9250.o
  Compiling out/src/lcd_st7920.o
  Compiling out/src/lcd_hd44780.o
  Compiling out/src/buttons.o
  Compiling out/src/tmcuart.o
  Compiling out/src/neopixel.o
  Compiling out/src/pulse_counter.o
  Compiling out/src/avr/main.o
  Compiling out/src/avr/timer.o
  Compiling out/src/avr/gpio.o
  Compiling out/src/avr/adc.o
  Compiling out/src/avr/spi.o
  Compiling out/src/avr/i2c.o
  Compiling out/src/avr/hard_pwm.o
  Compiling out/src/avr/watchdog.o
  Compiling out/src/avr/serial.o
  Compiling out/src/generic/serial_irq.o
  Building out/compile_time_request.o
Version: v0.11.0-86-g6026a99a
  Linking out/klipper.elf
  Creating hex file out/klipper.elf.hex
pi@speeder-pad:~/klipper$ ./scripts/update_mks_robin.py out/klipper.bin out/Robin_nano35.bin
Traceback (most recent call last):
  File "./scripts/update_mks_robin.py", line 38, in <module>
    main()
  File "./scripts/update_mks_robin.py", line 25, in main
    f = open(infilename, "rb")
IOError: [Errno 2] No such file or directory: 'out/klipper.bin'

**Comment by Guilouz** [comment, marked as answer]
You have not defined correct settings for firmware

**Reply by cyb1337** [reply]
This is what my settings look like, what is wrong here?
![image](https://user-images.githubusercontent.com/123260698/213878302-77745af8-7abe-4a8b-a6a6-e8698f327cf1.png)

**Reply by cyb1337** [reply]
oh im sorry I got it

**Reply by cyb1337** [reply]
For anyone else wondering, I oversaw that I also had to change the Microcontroller etc. as per the guide.

**Reply by NRICE88** [reply]
so i also oversaw the same thing but when i did everything as per guide (i hope) making sure all the settigns are like in the image the firmware still doesnt update when i start up the printer with the sd card in it, i may be missing something about that first part in the guide about the version am i supposed to check that and based on that information change something or?

**Reply by REPH26** [reply]
> so i also oversaw the same thing but when i did everything as per guide (i hope) making sure all the settigns are like in the image the firmware still doesnt update when i start up the printer with the sd card in it, i may be missing something about that first part in the guide about the version am i supposed to check that and based on that information change something or?

Did you find a solution ?

**Reply by Mike64p** [reply]
Bonjour,
Je n'ai plus accès à la photo pour la configuration correcte du firmware.
Possible de la remettre?
Merci d'avance.

**Comment by tkocou** [comment]
Anyone else having issues,  Guilouz is correct. I was searching for an answer as to why my Nano V2.1 controller would not talk to the speeder-pad. Turns out that I was not paying attention to the details laid out in the wiki page. Once I corrected my mistakes and recompiled the klipper.bin file, I followed the wiki page instructions and success.

**Comment by tkocou** [comment]
[https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Update-V400-Motherboard-Firmware-(Nano-V2.1-Clone---GD32F303)](url) no longer shows the screen shot images. It will be hard to follow the page instructions without the images.
